parcels/grid.py

```diff
@@ -78,7 +79,16 @@ def from_netcdf(cls, filenames, variables, dimensions, **kwargs):
             fields[var] = Field.from_netcdf(var, dimensions, dsets, **kwargs)
         u = fields.pop('U')
         v = fields.pop('V')
-        return cls(u, v, u.depth, u.time, fields=fields)
+
+        # assign time_origin if the time dimension has units and calendar
+        try:
+            time_units = dsets[0][dimensions['time']].units
+            calendar = dsets[0][dimensions['time']].calendar
+            time_origin = num2date(0, time_units, calendar)
```

Comment:
We are now assuming in a couple of places (another being in the `particle.py` constructor) that the time dimension from one field is the same for all. This is usually the U field, but here it's whatever is the first `dset`.

Is it time that we put some validation code in to ensure that `time` is consistent across fields, or if that does not need to be forced, that the user can specify the "time-origin" field? I had some odd results recently that took me a couple of days to track down because the time dimension from one field was at a different scale to another...